release-notes.txt

```diff
@@ -1,3 +1,536 @@
++ pkg/wakaama: implement IPSO sensor objects in LwM2M (#20652)
++ pkg/wamr: Added support for THUMB_VFP in wamr Makefile (#20628)
+* Makefile.include: creation of CACHEDIR.TAG as a dependency of pkg-
+  prepare (#20689)
+* pkg/littlefs2: bump to v2.9.2 (#20591)
+* pkg/tflite-micro: Update tflite-micro to latest version. requires
+  update of pkg/flatbuffers as well. (#20683)
+* pkg/wamr: updated package to WAMR 1.3.3 (#20691)
+
+Boards
+------
+
++ boards/arduino-nano-33-ble-sense: add support for arduino-
+  nano-33-ble-sense (#20668)
++ boards/b-l072z-lrwan1: add missing adc feature (#20733)
++ boards/nucleo-f303ze: add pinout (#20717)
++ boards/nucleo-l452re: add ADC (#20778)
++ boards/sodaq-sara-sff: add status pin for SARA (#20279)
++ cpu/stm32: add ADC support for WB55 (#20773)
++ drivers/led: add LED_NUMOF and convenience inline functions (#20783)
++ drivers/ws281x: Add saul support (#20562)
+* boards/common/stm32: simplify ifdef logic (#20617)
+* boards/nRF52xx: correctly set JLINK_DEVICE (#20599)
+* boards/nucleo-l476rg & stm32l476g-disco: fix MCU table (#20751)
+* boards/stm32l0538-disco: enable cpy2remed (#20766)
+* boards/stm32l476g-disco: enable cpy2remed (#20734)
+* build system: clean up netif features (#20700)
+* cpu/{gd32v,stm32}/periph/adc: make ADC clock setable (#19630)
+
+CPU
+---
+
++ cpu/cortexm: add stack limit support for Cortex-M33 (#20633)
+* cpu/cortexm: rework bkpt instruction call on panic (#20616)
+* cpu/cortexm_common: print last active thread on stack corruption (#20709)
+* cpu/esp: Handle format print errors (#20596)
+* cpu/msp430: implement power management (#20613)
+* cpu/nrf5x: implement pm_off() for nRF53/9160 (#20590)
+* cpu/nrf5x_common: fix ztimer issue on warm-boot (#20665)
+* cpu/nrf5x_common: properly calibrate RC-based low-frequency clock (#20669)
+* cpu/sam0: Remove deprecated muxpos (#20525)
+* cpu/sam0_common: avoid bitfield usage (#20747)
+* cpu/samd21: avoid the use of bitfield (#20713)
+* cpu/samd5x: avoid the use of bitfields (#20714)
+* cpu/samd5x: handle CAN errors (#20667)
+* cpu/saml1x: avoid the use of bitfield in register calls (#20710)
+* cpu/saml21: avoid the use of bitfield in register call (#20708)
+* cpu/stm32/periph/stm32_eth: provide confirm_send (#18428)
+* cpu/stm32: de-duplicate ifdefs (#20609)
+* periph/flashpage: remove deprecated flashpage_*_free functions (#18093)
+* pkg/lwip: make use of confirm send (#18427)
+* Remove extraneous bitwise assignment operation on NRF5X_common gpio.c (#20737)
+* sam0_eth: implement .confirm_send() to fix fragmented sending (#20666)
+* tree wide: compilation fixes for `native64` on musl systems (#20730)
+
+Device Drivers
+--------------
+
++ drivers/at24cxxx: Add M24C01 device and enhance documentation (#20588)
++ drivers/led: add LEDX_IS_PRESENT defines (#20637)
++ drivers/mtd: Add check for interger overflow (#20587)
++ Drivers: add MAX31855 thermocouple-to-digital converter (#20741)
+* drivers/at86rf215: port to new netdev API (#20672)
```

Comment:
This and the `stm32_eth` driver have been the first that have been converted. This may also be worth a note, as those are the first drivers that have been ported to the new API.

The blocker so far was lwIP, which has been stalling the conversion to the new API for ... two years or so?.

Support for the new netdev API has been added to lwIP in this release which allowed the first drivers to be converted.

There are two cool things this allows: The first is that the new API is easier to implement by drivers as the API more closely matches the behavior of real world hardware. The second is that the network stack now actually can synchronize with the netdev, which would allow maximizing TX utilization.

Reply:
Is there some place to read more about the differences between the old and the new netdev API? I've looked at the docs but couldn't find any.

Reply:
https://doc.riot-os.org/structnetdev__driver.html#a39585137953a92c1c58b5e0ad3262096 and https://doc.riot-os.org/structnetdev__driver.html#aacad0fea752b3a0ec9beb30470c4f34e explain both variants.

The plan back then was that netdevs are quickly converted with little fuzz, and backward compatibility with the old variant is dropped quickly after. But the PR to update lwIP to support both variants got stalled for years...

Anyway, with that resolved, it would indeed be nice to just quickly convert all netdevs to the new API.

Reply:
So maybe a good goal for the next release, and report success in the next release notes? :)

Reply:
Indeed. I hope I can find some time for that :)